app/controllers/provider_interface/provider_interface_controller.rb

```diff
@@ -1,7 +1,27 @@
 module ProviderInterface
   class ProviderInterfaceController < ActionController::Base
     include BasicAuthHelper
-    before_action :require_basic_auth_for_ui
+    before_action :authenticate_provider_user!
     layout 'application'
+
+    helper_method :current_user
+
+    def current_user
+      if session[:provider_user]
+        fake_user_class = Struct.new(:provider, :email_address)
+        fake_provider_class = Struct.new(:code)
+
+        fake_user_class.new(
+          fake_provider_class.new('ABC'),
+          session[:provider_user]['email_address'],
```

Comment:
`email_address` is a symbol when setting the session, can we use that there too?

Reply:
I've noticed some weird behaviour around `session` when working on similar code in `manage-courses-frontend`; I believe though it's set as a symbol and you can call it by a symbol within the current request lifecycle, it turns into a string on future requests. Could be mistaken, but I remember this being a bit unintuitive/tricky

Reply:
:+1: I'll see if I can make this consistent

Reply:
@tvararu is right. When the session contents are parsed Rails turns them into string keys, presumably for a good reason (all symbols are valid strings but not vice versa?) https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/middleware/session/cookie_store.rb#L80. 

I've added a commit to make the assignment use strings only, so it's now consistent.

Reply:
@tvararu is right. Rails turns session keys into strings on read: https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/middleware/session/cookie_store.rb#L80

I've changed the assignment to a string for consistency